File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        match = {0: 1}
        sum_ = ans = 0;
        for n in nums:
            sum_ += n
            ans += match.get(sum_ - k, 0)
            match[sum_] = match.get(sum_, 0) + 1
        return ans

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> match;
        match[0] = 1;
        int ans = 0;
        long sum_ = 0;
        int check;
        for(auto n: nums){
            sum_ += n;
            check = sum_ - k;
            if(match.find(check) != match.end()){
                ans += match[check];
            }
            ++match[sum_];
        }
        return ans;
    }
};